client/CPlayerInterface.cpp

```diff
@@ -446,6 +448,15 @@ void CPlayerInterface::heroKilled(const CGHeroInstance* hero)
 		adventureInt->selection = nullptr;
 }
 
+void CPlayerInterface::heroVisit(const CGHeroInstance * visitor, const CGObjectInstance * visitedObj, bool start)
+{
+	EVENT_HANDLER_CALLED_BY_CLIENT;
+	if(start && visitedObj->getVisitSound())
+	{
+		CCS->soundh->playSound(boost::str(boost::format("%s.wav") % visitedObj->getVisitSound().get()));

```

Comment:
I do not like this. Make get[...]Sound return proper value, or pre-process inside `playSound`.

Reply:
I also want to investigate how this would work with mods. Actually I think our filesystem able to play identical music even if it's has different extension / container, but I will need to test it.